tutorials/examples/test_graph_ring.py

```diff
@@ -0,0 +1,251 @@
+
+import math
+import time
+from typing import Optional
+import torch
+from torch import nn
+from gfn.actions import Actions, GraphActionType, GraphActions
+from gfn.gflownet.flow_matching import FMGFlowNet
+from gfn.gym import GraphBuilding
+from gfn.modules import DiscretePolicyEstimator
+from gfn.preprocessors import Preprocessor
+from gfn.states import GraphStates
+from tensordict import TensorDict
+from torch_geometric.nn import GCNConv
+import matplotlib.pyplot as plt
+
+
+def state_evaluator(states: GraphStates) -> torch.Tensor:
+    eps = 1e-6
+    if states.tensor["edge_index"].shape[0] == 0:
+        return torch.full(states.batch_shape, eps)
+    if states.tensor["edge_index"].shape[0] != states.tensor["node_feature"].shape[0]:
+        return torch.full(states.batch_shape, eps)
+
+    out = torch.zeros(len(states))
+    for i in range(len(states)):
+        start, end = states.tensor["batch_ptr"][i], states.tensor["batch_ptr"][i + 1]
+        edge_index_mask = torch.all(states.tensor["edge_index"] >= start, dim=-1) & torch.all(states.tensor["edge_index"] < end, dim=-1)
+        edge_index = states.tensor["edge_index"][edge_index_mask]
+        arange = torch.arange(start, end)
+        # TODO: not correct, accepts multiple rings
+        if torch.all(torch.sort(edge_index[:, 0])[0] == arange) and torch.all(torch.sort(edge_index[:, 1])[0] == arange):
+            out[i] = 1
+        else:
+            out[i] = eps
+    return out.view(*states.batch_shape)
+
+
+class RingPolicyEstimator(nn.Module):
+    def __init__(self, n_nodes: int, edge_hidden_dim: int = 128):
+        super().__init__()
+        self.action_type_conv = GCNConv(1, 1)
+        self.edge_index_conv = GCNConv(1, edge_hidden_dim)
+        self.n_nodes = n_nodes
+        self.edge_hidden_dim = edge_hidden_dim
+
+    def _group_sum(self, tensor: torch.Tensor, batch_ptr: torch.Tensor) -> torch.Tensor:
+        cumsum = torch.zeros((len(tensor) + 1, *tensor.shape[1:]), dtype=tensor.dtype, device=tensor.device)
+        cumsum[1:] = torch.cumsum(tensor, dim=0)
+        return cumsum[batch_ptr[1:]] - cumsum[batch_ptr[:-1]]
+
+    def forward(self, states_tensor: TensorDict) -> torch.Tensor:
+        node_feature = states_tensor["node_feature"].reshape(-1, 1)
+        edge_index = states_tensor["edge_index"].T
+        batch_ptr = states_tensor["batch_ptr"]
+
+        action_type = self.action_type_conv(node_feature, edge_index)
+        action_type = self._group_sum(action_type, batch_ptr)
+
+        edge_index = self.edge_index_conv(node_feature, edge_index)
+        edge_index = edge_index.reshape(*states_tensor["batch_shape"], self.n_nodes, self.edge_hidden_dim)
+        edge_index = torch.einsum("bnf,bmf->bnm", edge_index, edge_index)
+        edge_actions = edge_index.reshape(*states_tensor["batch_shape"], self.n_nodes * self.n_nodes)
+
+        return torch.cat([action_type, edge_actions], dim=-1)
+
+class RingGraphBuilding(GraphBuilding):
```

Comment:
I cannot run this file. I get this error
`ValueError: batch size was not specified when creating the TensorDict instance and it could not be retrieved from source.`

Can you run it?

Reply:
Yes, I can run it...
Which version of TensorDict are you using? I have `0.6.2`

Reply:
```
ERROR: Could not find a version that satisfies the requirement tensordict==0.6.2 (from versions: 0.0.1a0, 0.0.1b0, 0.0.1rc0, 0.0.2a0, 0.0.2b0, 0.0.3, 0.1.0, 0.1.1, 0.1.2, 0.2.0, 0.2.1, 0.3.0, 0.3.1, 0.3.2)
ERROR: No matching distribution found for tensordict==0.6.2
```

Althout in their github repo there is a 0.6.2 release. I will try to debug this.